bin/BuildPackages.sh

```diff
@@ -39,7 +49,9 @@ if [ -f gapicon.bmp ]
 fi
 
 # We need any subdirectory, to test if $GAPDIR is right
-SUBDIR=`ls -d */ | head -n 1`
+# dangerous to parse output of ls
+#SUBDIR=`ls -d */ | head -n 1`
+SUBDIR=`find -L ./ -maxdepth 1 -mindepth 1 -type d | sed 's|./||' | head -n 1`
```

Comment:
No, `find` is not a bash built-in (just type `help` or `help find` to get a list of built-in commands)

And `find -L ./` actually gives me dir names prefixed with `.//`, so with the grep, they still have a `/` prefix. 

But regardless of whether the prefix is `./` or `.//`, I don't see why it matters? There is no difference between `cd DIR` and `cd ./DIR` and `cd .//DIR` that I could tell. (Of course I still think that using `SUBDIR` is a HACK to start with, but we can clean that up later).

Finally, escaping: These are two different kinds of escaping :-). Using single quotes (versus double quotes, versus no quotes at all) affects how bash interprets inputs... But in fact, the string `|./||` contains no special characters and no spaces, hence the following three commands are all equialvalanet in bash:
```
sed 's|./||'
sed "s|./||"
sed "s|./||"
```
In each case, the `sed` executable sees as its first argument the same string.

However, the character `.` has a special meaning to sed respectively in regular expressions: It matches *any* character. So `./` matches not just `./` but also `x/` or `//` or ... luckily, in this case it usually won't matter -- unless perhaps a directory name contained a special character, which is unlikely *in this particular situation*. But I recommend coding defensively, instead of relying on unspoken assumptions, and thus sed should be told to match a period, not "an arbitrary" character. Which requires escaping on the level of sed, not on the level of bash. I.e.:
```
sed '|\./||'
```
But again, I think we can just not call `sed` at all here and it would be fine :)